code_generation/producer.py

```diff
@@ -129,13 +129,28 @@ def writecall(self, config, scope, shift=""):
         config[shift][scope]["df"] = "{df}"
         config[shift][scope]["vec_open"] = "{vec_open}"
         config[shift][scope]["vec_close"] = "{vec_close}"
+        # if a bool is used in the python configuration, convert it to a c++ bool value
+        # True -> true, False -> false
+        for para in config[shift][scope]:
+            if (
+                isinstance(config[shift][scope][para], bool)
+                and config[shift][scope][para]
+            ):
+                log.debug("Found a boolean True ! - converting to C++ syntax")
+                config[shift][scope][para] = "true"
+            if (
+                isinstance(config[shift][scope][para], bool)
+                and not config[shift][scope][para]
+            ):
+                log.debug("Found a boolean False ! - converting to C++ syntax")
+                config[shift][scope][para] = "false"
```

Comment:
It would be a bit shorter and I think a bit better understandible if you had the instance checked first and then the decision true/false inside.

Reply:
yeah true, it should be better now ..